packages/widgets/user-profile-widget/src/lib/widget/mixins/initMixin/initComponentsMixins/initUserCustomAttributesMixin.ts

```diff
@@ -0,0 +1,41 @@
+import { UserAttributeDriver } from '@descope/sdk-component-drivers';
+import { compose, createSingletonMixin } from '@descope/sdk-helpers';
+import { loggerMixin } from '@descope/sdk-mixins';
+import { getUserCustomAttrs } from '../../../state/selectors';
+import { stateManagementMixin } from '../../stateManagementMixin';
+import { initWidgetRootMixin } from './initWidgetRootMixin';
+
+export const initUserCustomAttributesMixin = createSingletonMixin(
+  <T extends CustomElementConstructor>(superclass: T) =>
+    class UserCustomAttributesMixinClass extends compose(
+      stateManagementMixin,
+      loggerMixin,
+      initWidgetRootMixin,
+    )(superclass) {
+      customValueUserAttr: UserAttributeDriver;
+
+      #initCustomValueUserAttrs() {
+        const allCustomAttributesComponents = this.shadowRoot?.querySelectorAll(
+          'descope-user-attribute[data-id^="customAttributes."]',
+        );
+
+        Array.from(allCustomAttributesComponents).forEach((nodeEle) => {
+          const attrName = nodeEle.getAttribute('data-id');
+          const customAttrName = attrName.replace('customAttributes.', '');
+
+          const compInstance = new UserAttributeDriver(nodeEle, {
+            logger: this.logger,
+          });
+
+          const userCustomAttributesData = getUserCustomAttrs(this.state);
+          compInstance.value = userCustomAttributesData[customAttrName];
```

Comment:
What if we do not have this attribute set for the user?
We will show a component with `undefined` value
I think in this case we should hide it

Reply:
Right - in that case I would show an empty line (as the editor did add the field). and potentially the user could edit it.
I would change to empty string in that case.